src/components/HomePage/Forms/SelectInputGenre.tsx

```diff
@@ -1,39 +1,50 @@
-import React, { FC } from 'react';
+import React, { FC, useRef } from 'react';
 import {
-  FormGroup, Input, Label,
+    Input, Label,
 } from 'reactstrap';
 import './Form.scss';
+import { Field, useField } from "formik";
 
-export const SelectInputGenre: FC<{ defaultSelected?: string[],
-    onChangeCallBack: Function }> = ({ defaultSelected, onChangeCallBack }) => {
-  const movieGenres = ['Action', 'Adventure', 'Animation',
-    'Comedy', 'Drama', 'Family',
-    'Fantasy', 'Horror', 'Music',
-    'Mystery', 'Romance', 'Science Fiction',
-    'Thriller'];
+export const SelectInputGenre: FC<{
+    defaultSelected: string[],
+    id: string
+}> = ({ defaultSelected, id }) => {
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const movieGenres = ['Action', 'Adventure', 'Animation',
+        'Comedy', 'Crime', 'Documentary',
+        'Drama', 'Family', 'Fantasy',
+        'Horror', 'Music', 'Mystery',
+        'Romance', 'Science Fiction', 'Thriller'];
+    const genresName = "genres";
+    const [form, meta, field] = useField(genresName);
 
-  return (
-    <FormGroup>
-      <Label for="movieGenre">
-        Genre
-      </Label>
-      <Input
-        id="movieGenre"
-        name="select"
-        type="select"
-        bsSize="sm"
-        multiple
-        onInput={e => {
-            const selectedOptions = Array.from((e.target as HTMLSelectElement).selectedOptions);
-            onChangeCallBack(selectedOptions.map(item => item.text));
-        }
-        }
-      >
-        <option value="" disabled className="display:none;">Select Genre</option>
-        {movieGenres.map((genre) => (
-          <option key={genre} selected={defaultSelected?.includes(genre)}>{genre}</option>
-        ))}
-      </Input>
-    </FormGroup>
-  );
+    return (
+        <Field name={genresName}>
+            {() => (
+                <div>
+                    <Label for={id}>
+                        Genre
+                    </Label>
+                    <Input
+                        id={id}
+                        name="select"
+                        type="select"
+                        bsSize="sm"
+                        multiple
+                        innerRef={inputRef}
+                        onInput={e => {
+                            const selectedOptions = Array.from((e.target as HTMLSelectElement).selectedOptions);
```

Comment:
Let's extract to function, e.g. :
`const handleOnInput = (e) => {...}`

Reply:
Fixed it in commit 144e7682797a8b327df9c556453f21aad1d10993